From 894a06baa3943598523491b26902571fd5d5d295 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Thu, 13 Nov 2014 21:35:39 +0300
Subject: [PATCH 209/213] version 1.2: Event dates

Dates for birth and dry-off events can go up to 2 years and 6 months
 back respectively

Signed-off-by: Jason Rogena <j.rogena@cgiar.org>
---
 NgombePlanner/src/main/AndroidManifest.xml           |  4 ++--
 .../org/cgiar/ilri/np/farmer/AddEventActivity.java   |  6 +++++-
 .../ilri/np/farmer/CowRegistrationActivity.java      | 20 +++++++++++---------
 .../cgiar/ilri/np/farmer/backend/DataHandler.java    |  3 ++-
 4 files changed, 20 insertions(+), 13 deletions(-)

diff --git a/NgombePlanner/src/main/AndroidManifest.xml b/NgombePlanner/src/main/AndroidManifest.xml
index 53e4246..f6eb866 100644
--- a/NgombePlanner/src/main/AndroidManifest.xml
+++ b/NgombePlanner/src/main/AndroidManifest.xml
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.np.farmer"
-    android:versionCode="23"
-    android:versionName="1.1.2" >
+    android:versionCode="25"
+    android:versionName="1.2" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/AddEventActivity.java b/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/AddEventActivity.java
index 2bddfb8..f77a3ca 100644
--- a/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/AddEventActivity.java
+++ b/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/AddEventActivity.java
@@ -655,7 +655,7 @@ public class AddEventActivity extends SherlockActivity implements NPActivity, Vi
                 Toast.makeText(this,dateInFuture,Toast.LENGTH_LONG).show();
                 return false;
             }
-            else if(days > 15 && !eventTypesEN[eventTypeS.getSelectedItemPosition()].equals("Birth")) {//more than 15 days for other events other than birth
+            else if(days > 15 && !eventTypesEN[eventTypeS.getSelectedItemPosition()].equals("Birth") && !eventTypesEN[eventTypeS.getSelectedItemPosition()].equals("Dry Off")) {//more than 15 days for other events other than birth and dry off
                 Toast.makeText(this,Locale.getStringInLocale("event_too_old",this),Toast.LENGTH_LONG).show();
                 return false;
             }
@@ -663,6 +663,10 @@ public class AddEventActivity extends SherlockActivity implements NPActivity, Vi
                 Toast.makeText(this,Locale.getStringInLocale("event_too_old",this),Toast.LENGTH_LONG).show();
                 return false;
             }
+            else if(days > 180 && eventTypesEN[eventTypeS.getSelectedItemPosition()].equals("Dry Off")) {//more than 6 months for dry offs
+                Toast.makeText(this,Locale.getStringInLocale("event_too_old",this),Toast.LENGTH_LONG).show();
+                return false;
+            }
             else {
                 return true;
             }
diff --git a/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/CowRegistrationActivity.java b/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/CowRegistrationActivity.java
index 8480e80..d822fd1 100644
--- a/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/CowRegistrationActivity.java
+++ b/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/CowRegistrationActivity.java
@@ -526,15 +526,17 @@ public class CowRegistrationActivity extends SherlockActivity implements NPActiv
                     if(thisCow.getMode().equals(Cow.MODE_BORN_CALF_REGISTRATION)) {
                         this.setTitle(Locale.getStringInLocale("calf_registration",this));
                         setAgeFromDate(thisCow.getDateOfBirth());
-                        thisCow.setAge(Integer.parseInt(ageET.getText().toString()));
-                        if(ageTypesInEN[ageS.getSelectedItemPosition()].equals("Days")) {
-                            thisCow.setAgeType(Cow.AGE_TYPE_DAY);
-                        }
-                        else if(ageTypesInEN[ageS.getSelectedItemPosition()].equals("Months")) {
-                            thisCow.setAgeType(Cow.AGE_TYPE_MONTH);
-                        }
-                        else if(ageTypesInEN[ageS.getSelectedItemPosition()].equals("Years")) {
-                            thisCow.setAgeType(Cow.AGE_TYPE_YEAR);
+                        if(ageET.getText().toString().length() > 0){
+                            thisCow.setAge(Integer.parseInt(ageET.getText().toString()));
+                            if(ageTypesInEN[ageS.getSelectedItemPosition()].equals("Days")) {
+                                thisCow.setAgeType(Cow.AGE_TYPE_DAY);
+                            }
+                            else if(ageTypesInEN[ageS.getSelectedItemPosition()].equals("Months")) {
+                                thisCow.setAgeType(Cow.AGE_TYPE_MONTH);
+                            }
+                            else if(ageTypesInEN[ageS.getSelectedItemPosition()].equals("Years")) {
+                                thisCow.setAgeType(Cow.AGE_TYPE_YEAR);
+                            }
                         }
                     }
                 }
diff --git a/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/backend/DataHandler.java b/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/backend/DataHandler.java
index de91390..3261840 100644
--- a/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/backend/DataHandler.java
+++ b/NgombePlanner/src/main/java/org/cgiar/ilri/np/farmer/backend/DataHandler.java
@@ -71,7 +71,8 @@ public class DataHandler
     private static final int HTTP_POST_TIMEOUT =20000;
     private static final int HTTP_RESPONSE_TIMEOUT =20000;
     public static final String SMS_SERVER_ADDRESS = "+254708090206";
-    private static final String BASE_URL="http://azizi.ilri.cgiar.org/ngombe_planner";
+    private static final String BASE_URL="http://178.79.156.47/ngombe_planner";
+    //private static final String BASE_URL="http://azizi.ilri.cgiar.org/ngombe_planner";
     //private static final String BASE_URL="http://192.168.14.102/~jason/ngombe_planner/WebServer";
     //private static final String BASE_URL="http://172.26.23.48/~jason/ngombe_planner/WebServer";
     public static final String FARMER_REGISTRATION_URL="/php/farmer/registration.php";
-- 
2.7.1

